### ISSUE TITLE
Shelly 1 - WiFi not reconnecting after router reboot

### ISSUE THREAD

**Issue by SimoHDK** [mongoose-os-apps/shelly-homekit#1107]
Hi All,

I believe the bug around wifi not reconnecting still exists, as it seems to be impacting my Shelly1 with latest firmware.

Scenario:
1. Shelly is conncected and works well (Low RSSI however, around ~79)
2. Router will reboot due to power outage or similar (Shelly typically remains ON during this time as it's on battery backup)
3. Router is online, Shelly won't re-connect to wifi
4. I login to Shelly directly (I leave AP mode ON for this purpose) and always find the 'Enable wifi' toggle switched OFF.
5. Toggling this back to ON and WiFi reconnects, no problems.

Any suggestions?

**Comment by pboudoin** [comment]
Same issue here, and also the same observation

**Comment by MBarti** [comment]
Yup, same here. Shelly Plus 1 set as lock. Loses availability in Homekit (no response), but is available through the web interface. 
My setup: Ubiquiti network, strong signal, fixed IP, several Homepod Minis and Apple TV HD; firmware 2.11.2

**Comment by timoschilling** [comment]
@MBarti can you please make a screenshot from what you see in the Home app.

**Comment by MBarti** [comment]
Sure. I simply see "no response" and the icon is grayed out. When clicking to expand the accessory, it says "Accessory is not responding".

<img width="496" alt="image" src="https://user-images.githubusercontent.com/22871912/207348664-3a0f93a3-5b1d-4b6c-bc75-533a53d732a3.png">
<img width="518" alt="image" src="https://user-images.githubusercontent.com/22871912/207348783-a9260ae9-11a5-4603-a385-b7bd8460983f.png">

**Comment by gerhardwallnoefer** [comment]
Exactly the same for me. Shelly 2.5 with firmware 2.11.2

**Comment by just-doit** [comment]
I probably have a similar problem. What I can not say whether it is related to a router reboot.

The device is a Shelly1 with the following firmware: `20220812-225235/2.11.2-g1b146b4-master`.

However, the switch has now lost connection to the Access Point for the second time in two days. I had to restart the device (turn off / turn on the power at the breaker)
It may be related to the fact that this Shelly1 has two APs within range.
I have a second Shelly1, but it only sees one of the two APs.
Powersaving is disabled and RRSI is around -62, if this is relevant. ;-)
I have now activated the AP mode, so that I can possibly still get to the device when it is offline again. 
Then I'll see what the log says and post it here.

**Comment by d4rkmen** [comment]
Have Shelly 1 (as a garage door opener) running for a long time and not happen this issue

**Comment by pm0u** [comment]
Just had this issue on a shelly Plus1 - AP mode was turned off so I could not diagnose any further. Router was rebooted as I flipped some breakers around the house but Shelly was untouched. Router back online, Shelly did not reconnect. Rebooted Shelly, it reconnected but was in OFF state when it was last in ON state while not connected. Turned Shelly on. Back to normal. A little concerning as this is a furnace control and I would not want to to ever boot in OFF

**Comment by kuraz758** [comment]
same here. it seems that whenever i mess with my router and it loses connection, it never reconnects

**Comment by Claudio19876** [comment]
same problem

**Comment by mathieucarbou** [comment]
Same issue but in my case I have a mesh system router + satellite.

I reboot the satellite, and all the Shelly Plus 16A that were connected to the satellite won't reconnect to it after it is rebooted and synced.

**Comment by markirb** [comment]
No easy answer for this one. But maybe an easy question: does this happen on stock fw also?

**Comment by kuraz758** [comment]
yes, i only ever used stock fw

Markus Kirberg ***@***.***> schrieb am So., 9. Juni 2024,
06:49:

**Comment by MBarti** [comment]
No, never happened on stock fw and I have over 20 Shellys.

**Comment by saoron** [comment]
What worked for me is to calibrate the device after the flash to homekit fw.

An error message is being sent (I found it via homebridge while trying to identify the issue)
<img width="931" alt="image" src="https://github.com/mongoose-os-apps/shelly-homekit/assets/4144775/5155a4f2-c392-4ccc-9b3d-ed801d68cabc">

**Comment by kuraz758** [comment]
I'm sorry, i didn't even realize that this is about an alternative firmware

Yossi Neiman ***@***.***> schrieb am Mi., 12. Juni 2024,
13:48:
